__tests__/pages/api/companies/[id]/archiveCompany.test.ts

```diff
@@ -0,0 +1,146 @@
+import { prisma } from '@/lib/prisma'
+import handler from '@/pages/api/companies/[id]'
+import {
+  cleanDatabase,
+  mockDELETERequest,
+  parseJSON,
+} from '../../../../helpers'
+import type { Company } from '@prisma/client'
+
+const ENDPOINT = '/api/companies/[id]'
+
+describe(`DELETE ${ENDPOINT}`, () => {
+  beforeEach(async () => {
+    await cleanDatabase(prisma)
+  })
+
+  describe('when a valid company is successfully archived', () => {
+    it('returns HTTP 200 and success message', async () => {
+      const company: Company = await prisma.company.create({
+        data: {
+          display_name: 'Test Company',
+          comment: 'This is a test company',
+        },
+      })
+
+      const { req, res } = mockDELETERequest({ id: company.id })
+
+      await handler(req, res)
+
+      expect(res._getStatusCode()).toBe(200)
+      expect(parseJSON(res)).toEqual({
+        success: true,
+        message: 'Company archived successfully',
+      })
+
+      const archivedCompany = await prisma.company.findUnique({
```

Comment:
Actually we don't need to check status changing inside the database, because controller tests (or feature tests) shouldn't know anything about database layer. They just check request/response flow.